server/src/test/java/org/spine3/server/aggregate/AggregateRootShould.java

```diff
@@ -85,55 +137,91 @@ private ProjectRoot(BoundedContext boundedContext, ProjectId id) {
+            return result;
         }
 
         @Apply
         private void apply(ProjectCreated event) {
             getBuilder().setId(event.getProjectId())
                         .setName(event.getName());
         }
+    }
 
-        @Assign
-        public TaskAdded handle(AddTask msg, CommandContext context) {
-            return TaskAdded.newBuilder()
-                            .setProjectId(msg.getProjectId())
-                            .build();
-        }
+    @SuppressWarnings("AssignmentToStaticFieldFromInstanceMethod")
+    // Static field in the instance method is used for the test simplification.
+    private static class ProjectLifeCyclePart
+            extends AggregatePart<ProjectId, ProjectLifecycle, ProjectLifecycle.Builder> {
 
-        @Apply
-        private void apply(TaskAdded event) {
-            getBuilder().setId(event.getProjectId());
+        private static boolean exceptionOccurred = false;
+
+        protected ProjectLifeCyclePart(ProjectId id, ProjectRoot root) {
+            super(id, root);
         }
 
         @Assign
-        public ProjectStarted handle(StartProject msg, CommandContext context) {
-            return ProjectStarted.newBuilder()
-                                 .setProjectId(msg.getProjectId())
-                                 .build();
+        ProjectStarted handle(StartProject msg) {
+            final ProjectDefinition projectDefinition = getPartState(ProjectDefinition.class);
+            final boolean defaultId = projectDefinition.getId()
+                                                       .equals(ProjectId.getDefaultInstance());
+            if (defaultId) {
+                exceptionOccurred = true;
+            }
+            final ProjectStarted result = ProjectStarted.newBuilder()
+                                                        .setProjectId(msg.getProjectId())
+                                                        .build();
+            return result;
         }
 
         @Apply
         private void apply(ProjectStarted event) {
+            getBuilder().setStatus(Status.STARTED);
+        }
+    }
+
+    private static class ProjectDefinitionRepository
+            extends AggregatePartRepository<ProjectId, ProjectDefinitionPart, ProjectRoot> {
+
+        private ProjectDefinitionRepository(BoundedContext boundedContext) {
+            super(boundedContext, ProjectRoot.class);
+        }
+    }
+
+    private static class ProjectLifeCycleRepository
+            extends AggregatePartRepository<ProjectId, ProjectLifeCyclePart, ProjectRoot> {
+
+        private ProjectLifeCycleRepository(BoundedContext boundedContext) {
+            super(boundedContext, ProjectRoot.class);
         }
     }
 
-    private static class ProjectHeaderRepository extends AggregatePartRepository<ProjectId, ProjectHeader> {
+    private static class MockStreamObserver implements StreamObserver<Response> {
+        @Override
+        public void onNext(Response value) {
+
```

Comment:
Please document empty method bodies or (assuming this is a mock object) just remove empty lines.